instrumentation/opentelemetry-instrumentation-grpc/tests/test_aio_client_interceptor_hooks.py

```diff
@@ -0,0 +1,94 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+try:
+    from unittest import IsolatedAsyncioTestCase
+except ImportError:
+    # unittest.IsolatedAsyncioTestCase was introduced in Python 3.8. It's use
+    # simplifies the following tests. Without it, the amount of test code
+    # increases significantly, with most of the additional code handling
+    # the asyncio set up.
+    from unittest import TestCase
+
+    class IsolatedAsyncioTestCase(TestCase):
+        def run(self, result=None):
+            self.skipTest(
+                "This test requires Python 3.8 for unittest.IsolatedAsyncioTestCase"
+            )
+
+import grpc
+import pytest
+
+from opentelemetry.instrumentation.grpc import (
+    GrpcAioInstrumentorClient,
+    aio_client_interceptors,
+    filters,
+)
+from opentelemetry.test.test_base import TestBase
+
+from ._aio_client import (
+    simple_method,
+)
+from ._server import create_test_server
+from .protobuf import test_server_pb2_grpc  # pylint: disable=no-name-in-module
+
+
+@pytest.mark.asyncio
+class TestAioClientInterceptorWithHooks(TestBase, IsolatedAsyncioTestCase):
+    def setUp(self):
+        super().setUp()
+
+        def request_hook(span, request):
+            span.set_attribute("request_data", request.request_data)
+
+        def response_hook(span, response):
+            span.set_attribute("response_data", response)
+
+        GrpcAioInstrumentorClient().instrument(
+                request_hook=request_hook,
+                response_hook=response_hook,
+            )
+        self.server = create_test_server(25565)
+        self.server.start()
+
+        self._channel = grpc.aio.insecure_channel(
+            "localhost:25565",
+        )
+
+        self._stub = test_server_pb2_grpc.GRPCTestServerStub(self._channel)
+
+    def tearDown(self):
+        super().tearDown()
+        GrpcAioInstrumentorClient().uninstrument()
+        self.server.stop(None)
+        self._channel.close()
+
+    async def test_request_and_response_hooks(self):
```

Comment:
Please add a text that checks a case when the hook throws an exception

Reply:
sababaaa i'll add a test and see that this doesn't crash :)